File: PUGD-frontend/graphql/mutations/acquisition/orderline.js
import gql from "graphql-tag";

const InsertOrderLine = gql`
  mutation(
    $_id: String!
    $order: String!
    $isbn: String!
    $title: String
    $author: String
    $quantity: Int
    $price: Float
    $discount: Float
    $status: String
  ) {
    insertOrderLine(
      _id: $_id
      order: $order
      isbn: $isbn
      title: $title
      author: $author
      quantity: $quantity
      price: $price
      discount: $discount
      status: $status
    )
  }
`;

const UpdateOrderLine = gql`
  mutation(
    $_id: String!
    $isbn: String
    $title: String
    $author: String
    $quantity: Int
    $price: Float
    $discount: Float
    $status: String
  ) {
    updateOrderLine(
      _id: $_id
      isbn: $isbn
      title: $title
      author: $author
      quantity: $quantity
      price: $price
      discount: $discount
      status: $status
    )
  }
`;

const DeleteOrderLine = gql`
  mutation($_id: String!) {
    deleteOrderLine(_id: $_id)
  }
`;

//important
module.exports = {
  InsertOrderLine,
  UpdateOrderLine,
  DeleteOrderLine,
};

File: PUGD-api/graphql/types/AcquisitionTypes/OrderLine.go
package AcquisitionTypes

import "github.com/graphql-go/graphql"

var OrderLineType = graphql.NewObject(graphql.ObjectConfig{
	Name: "OrderLineType",
	Fields: graphql.Fields{
		"_id": &graphql.Field{
			Type: graphql.ID,
		},
		"order": &graphql.Field{
			Type: graphql.ID,
		},
		"isbn": &graphql.Field{
			Type: graphql.String,
		},
		"title": &graphql.Field{
			Type: graphql.String,
		},
		"author": &graphql.Field{
			Type: graphql.String,
		},
		"quantity": &graphql.Field{
			Type: graphql.Int,
		},
		"quantityreceived": &graphql.Field{
			Type: graphql.Int,
		},
		"quantityfactured": &graphql.Field{
			Type: graphql.Int,
		},
		"price": &graphql.Field{
			Type: graphql.Float,
		},
		"discount": &graphql.Field{
			Type: graphql.Float,
		},
		"status": &graphql.Field{
			Type: graphql.String,
		},
	},
})
